packages/core/useEventSource/index.ts

```diff
@@ -110,16 +110,16 @@ function resolveNestedOptions<T>(options: T | true): T {
  * @param events
  * @param options
  */
-export function useEventSource<Events extends string[]>(
+export function useEventSource<Events extends string[], Data = any>(
   url: MaybeRefOrGetter<string | URL | undefined>,
   events: Events = [] as unknown as Events,
   options: UseEventSourceOptions = {},
 ): UseEventSourceReturn<Events> {
-  const event: Ref<string | null> = deepRef(null)
-  const data: Ref<string | null> = deepRef(null)
-  const status = shallowRef('CONNECTING') as Ref<EventSourceStatus>
-  const eventSource = deepRef(null) as Ref<EventSource | null>
-  const error = shallowRef(null) as Ref<Event | null>
+  const event: ShallowRef<string | null> = shallowRef(null)
+  const data: ShallowRef<Data | null> = shallowRef(null)
+  const status = shallowRef<EventSourceStatus>('CONNECTING')
+  const eventSource = deepRef<EventSource | null>(null)
+  const error = shallowRef<Event | null>(null)
```

Comment:
I think this can be done like this: https://github.com/vueuse/vueuse/pull/4591

So we do not actually break deep refs on data.

Reply:
I'd personally not bother as it clearly should be shallow. We need to decide when to break these to make them that way

Otherwise we will end up with this deep flag workaround everywhere, and it'll be even more difficult to get rid of in future

Now is our chance to fix this and it'll be less breaking than removing the deep flag one day

Reply:
From discord discussion with @43081j:

TL;DR:

- returned immutable data should be `readonly` and `shallow`
- complex state that should not be `readonly` should default to shallow and have an opt-out (`deepRefs`) (rare)
- exceptions always exist